src/core/algorithms/ucc/ucc_verifier/ucc_verifier.h

```diff
@@ -0,0 +1,65 @@
+#pragma once
+
+#include <vector>
+
+#include "algorithms/algorithm.h"
+#include "config/equal_nulls/type.h"
+#include "config/indices/type.h"
+#include "config/tabular_data/input_table_type.h"
+#include "model/table/column_layout_relation_data.h"
+
+namespace algos {
+
+/* Algorithm used for verify that a set of columns is UCC and retrieving useful information in
+ * case it is not */
+class UCCVerifier : public Algorithm {
+private:
+    /* input options */
+    config::IndicesType column_indices_;
+    config::EqNullsType is_null_equal_null_{};
+
+    config::InputTable input_table_;
+    std::shared_ptr<ColumnLayoutRelationData> relation_;
```

Comment:
```suggestion
    std::unique_ptr<ColumnLayoutRelationData> relation_;
```
There is no reason to use `shared_ptr` here